CMakeLists.txt

```diff
@@ -55,7 +56,6 @@ set(PYBIND11_HEADERS
   include/pybind11/pytypes.h
   include/pybind11/stl.h
   include/pybind11/stl_bind.h
-  include/pybind11/typeid.h
```

Comment:
My thinking here was that `typeid.h` is included anyway by `cast.h` and in turn by `pybind11.h`, so there's no way to lose out on `py::type_id<T>()` unless someone is using it without anything else from pybind11. But I wouldn't mind reverting it if I'm overlooking something.